service/framework.go

```diff
@@ -175,15 +177,53 @@ func newCRDFramework(config Config) (*framework.Framework, error) {
 		}
 	}
 
-	var crdFramework *framework.Framework
+	handlesFunc := func(obj interface{}) bool {
+		return true
+	}
+
+	initCtxFunc := func(ctx context.Context, obj interface{}) (context.Context, error) {
+		return ctx, nil
+	}
+
+	var v2ResourceSet *framework.ResourceSet
 	{
-		c := framework.DefaultConfig()
+		c := framework.ResourceSetConfig{
 
-		c.CRD = v1alpha1.NewIngressConfigCRD()
-		c.CRDClient = crdClient
-		c.Informer = newInformer
-		c.Logger = config.Logger
-		c.ResourceRouter = framework.DefaultResourceRouter(resources)
+			Handles:   handlesFunc,
+			InitCtx:   initCtxFunc,
+			Logger:    config.Logger,
+			Resources: resources,
+		}
+
+		v2ResourceSet, err = framework.NewResourceSet(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var resourceRouter *framework.ResourceRouter
+	{
+		c := framework.ResourceRouterConfig{}
+
+		c.ResourceSets = []*framework.ResourceSet{
+			v2ResourceSet,
+		}
+
+		resourceRouter, err = framework.NewResourceRouter(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var crdFramework *framework.Framework
+	{
+		c := framework.Config{
+			CRD:            v1alpha1.NewIngressConfigCRD(),
+			CRDClient:      crdClient,
+			Informer:       newInformer,
+			Logger:         config.Logger,
+			ResourceRouter: resourceRouter,
+		}
```

Comment:
This is just to make it work as it is while updating the deps and use `dep`. More coming soon.